### PR TITLE
Issue/521/upgrade vue to version 3

### LINKED ISSUE #521: Upgrade Vue to version 3
More time needed for:
- [ ] vue-concise-slider - lack of package version for vue 3
- [ ] vee-validate - totally different API

### PR DESCRIPTION
**Pull request by dbetka**
close #521 

Subtasks:
- #544 

Important after merge to devel:
- Introduce to new Vue.js devtools
- List errors possible to appear after merge (click inheritance)

What done: 
- Upgrade packages to next version and adapt code for it: 
  - vue, 
  - vue-i18n,
  - vue-router, 
  - vuex, 
  - vue-loader, 
  - @vue/compile-sfc
  - @dbetka/vue-material-icons
  - vue-ellipse-progress
  - vue3-touch-events
  - openlayers
- Replace beforeDestroy with beforeUnmount
- Fix Vue.js devtools configuration and add instruction for it
- Remove unnecessary click emit inheritance
- Update eslint configuration
- Bump to version 2.0.0-dev
- Use vue3-flicking instead vue-concise-slider for user guide carousel

### CODE REVIEW

File: client/src/router/index.js [hunk 1]
```diff
@@ -11,32 +10,30 @@ import { guardsUtils } from 'src/router/guards';
...
-      if (Router.isNavigationFailure(err, Router.NavigationFailureType.duplicated)) {
-        // resolve err
-        return err;
-      }
-      // rethrow error
-      return Promise.reject(err);
-    });
-  };
-})();
+// (function silenceNavigationFailureErrors () {
```
Comment (Bedpaw) [inline client/src/router/index.js:19]: Why remove? If remove necessery why comment?

**Comment by Bedpaw** [comment]
Found problems:
- [ ]  outlined icons doesn't work
- [ ] buttons in events-list emit click x4
- [ ] buttons in user popup emit click x2
- [ ] user popup and other X symbol not closing popup

File: client/src/router/index.js (hunk 1 above)
Comment (dbetka) [reply client/src/router/index.js:19]: Because this solution is not working in Vue 3.
I'll find other solution for it.

Comment (dbetka) [reply client/src/router/index.js]: I've tested and console has no any errors without `silenceNavigationFailureErrors`.
I'll remove that code.

**Pull request merged**